### PR TITLE
Refresh admin panel for bulk actions and filling out missing features

### PR DESCRIPTION
**Pull request by JoshAshby**
Most of this is just some niceness, part of it is because I need to make a test user for submitting with the Firefox extension, and I don't want to send myself an invite. So, as a result, I started down the path of making a basic "new user" admin page and some tools to help manage users like resetting passwords (since they can't do that on their own for now) and grew into just committing to filling out most of the currently missing features.

Adds:
 - Basic search & filtering of admin lists
 - Bulk actions using the new JS helpers
 - Few new `show` views
 - Few new forms

### CODE REVIEW

File: app/views/admin/service_announcements/_menu.html.haml [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
+  .item
+    .ui.transparent.icon.input
+      = form_tag admin_service_announcements_path, method: :get do
+        = search_field_tag "q", params[:q], class: "ui prompt", placeholder: "Search ...", aria: { label: "search" }
+      %i.search.link.icon
 
   = active_link_to admin_service_announcements_path, active: :exclusive, class: "item" do
     Index
-    - if count
+    - if defined?(count) && count
```
Comment (robacarp) [inline app/views/admin/service_announcements/_menu.html.haml]: this is weird

File: app/views/admin/service_announcements/show.html.haml [hunk 2]
```diff
@@ -1,71 +1,85 @@
...
+.ui.menu
+  = link_to edit_admin_service_announcement_path(@service_announcement), class: "item" do
+    %i.edit.icon
+    edit
 
-    .right.menu
-      .item{ data: { behavior: "modal", template: "admin/templates/service_announcements/delete", url: admin_service_announcement_path(@service_announcement), title: @service_announcement.title } }
-        %i.trash.icon
-        delete
+  = link_to admin_service_announcement_toggle_index_path(@service_announcement), class: "item", method: :post do
```
Comment (robacarp) [inline app/views/admin/service_announcements/show.html.haml]: dat route helper

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,10 @@
 require "sidekiq/web"
 
+def draw name
+  path = Rails.root.join "config", "routes", "#{ name }.rb"
+  instance_eval path.read, path.to_s
+end
```
Comment (robacarp) [inline config/routes.rb]: ❤️

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -116,4 +106,17 @@ def modal_tag model, template, **opts, &block
     tag.div(**options, &block)
   end
   # rubocop:enable Metrics/AbcSize
+
+  def template_path_for *args
+    template_prefix = "public" unless controller_path.start_with? "admin/"
+    template_prefix ||= "admin"
+
```
[the remark this answers is not in the document]
Comment (JoshAshby) [reply app/helpers/application_helper.rb]: I don't want to talk about it

File: app/views/admin/service_announcements/_menu.html.haml (hunk 1 above)
Comment (JoshAshby) [reply app/views/admin/service_announcements/_menu.html.haml]: Yeah :< But needed, if I just have it as the old `if count` then it fails with undefined method count if I don't pass it in through `locales: {}`

**Pull request merged**